app/windowsInit.js

```diff
@@ -2,109 +2,246 @@
-      !process.argv.includes('--relaunch') &&
-      !process.argv.includes('--user-data-dir-name=brave-development')) {
-    delete process.env.CHROME_USER_DATA_DIR
-    if (isSquirrelFirstRun) {
-      app.relaunch({args: [userDataDirSwitch, '--relaunch']})
-    } else {
-      app.relaunch({args: process.argv.slice(1).concat([userDataDirSwitch, '--relaunch'])})
+  const getBraveCoreInstallPath = () => {
+    const braveCoreInstallLocations = [
+      '%USERPROFILE%\\AppData\\Local\\BraveSoftware\\Brave-Browser\\Application',
+      '%ProgramFiles(x86)%\\BraveSoftware\\Brave-Browser\\Application',
+      '%ProgramFiles%\\BraveSoftware\\Brave-Browser\\Application'
+    ]
+
+    // check for existing installations
+    for (let i = 0; i < braveCoreInstallLocations.length; i++) {
+      const path = braveCoreInstallLocations[i]
+      const resolvedPath = path.replace(/%([^%]+)%/g, function (_, variableToResolve) {
+        return process.env[variableToResolve]
+      })
+      if (fs.existsSync(resolvedPath)) {
+        debugLog(`brave-core already installed at "${resolvedPath}"`)
+        return resolvedPath
+      }
     }
-    app.exit()
+
+    return false
   }
-}
 
-app.on('will-finish-launching', () => {
-  app.setAppUserModelId(appUserModelId)
-})
+  const installBraveCore = () => {
+    // get path to the bundled brave-core binary
+    const installerPath = getBraveCoreInstallerPath()
+    if (!installerPath) {
+      debugLog('brave-core installer not found')
+      return false
+    }
+
+    // forcefully create the legacy shortcut
+    const updateExe = path.join(getBraveBinPath(), '..', 'Update.exe')
+    const shortcutTarget = path.basename(process.execPath)
+    const shortcutCmd = `${updateExe} --createShortcut=${shortcutTarget} --shortcut-locations=DuplicateDesktop --process-start-args=--launch-muon`
+    try {
+      debugLog('Creating "Brave (old)" shortcut on desktop')
+      execSync(shortcutCmd)
+    } catch (e) {
+      debugLog('Error thrown when creating Muon shortcut: ' + e.toString())
+    }
+
+    // brave-core is not installed; go ahead with silent install
+    const installCmd = `${installerPath} /silent /install`
+    try {
+      debugLog(`Attempting silent install using "${installerPath}"`)
+      execSync(installCmd)
+    } catch (e) {
+      debugLog('Error thrown when installing brave-core: ' + e.toString())
+      return false
+    }
+
+    // store details to disk; no further install attempts will be made
+    try {
+      fs.writeFileSync(braveCoreUpgradeFile, `installed: ${new Date().getTime()}`)
+    } catch (e) {
+    }
+
+    // relaunch and append argument expected in:
+    // https://github.com/brave/brave-browser/issues/1545
+    try {
+      const installedPath = getBraveCoreInstallPath()
+      debugLog(`Launching brave-core at "${installedPath}/brave.exe" with "--upgrade-from-muon" flag`)
+      execSync(`"${installedPath}/brave.exe" --upgrade-from-muon`)
```

Comment:
Shouldn't this be a `\` instead of a `/`. I'm sure Windows will accept it though, so just a nit